CRM/MembersOnlyEvent/Service/MembersOnlyEventAccessService.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as EventMembershipType;
+use CRM_MembersOnlyEvent_Configurations as Configurations;
+use CRM_MembersOnlyEvent_BAO_MembersOnlyEvent as MembersOnlyEvent;
+
+class CRM_MembersOnlyEvent_Service_MembersOnlyEventAccessService {
+
+  /**
+   * @var array
+   */
+  private $contactActiveAllowedMemberships = [];
```

Comment:
if you want to avoid extra queries, maybe you can make $contactActiveAllowedMemberships and $membersOnlyEvent  static class properties, and thus they will be shared across all the instances of this class.